### PR TITLE
Date fixes

### PR DESCRIPTION
**Pull request by gknisely**
Clean up.
When a route crosses the beginning or end of dst, we should adjust the time correctly.
Added lots of tests.

### CODE REVIEW

**Comment by gknisely** [comment]
fixes #230

File: src/baldr/datetime.cc [hunk 1]
```diff
@@ -208,7 +207,7 @@ uint64_t remove_service_day(const uint64_t& days, const boost::gregorian::date&
 }
 
 // check if service is available for a date.
-bool is_service_available(const uint64_t& days, const uint32_t& start_date, const uint32_t& date, const uint32_t& end_date) {
+bool is_service_available(const uint64_t days, const uint32_t start_date, const uint32_t date, const uint32_t end_date) {
```
Comment (kevinkreiser) [inline src/baldr/datetime.cc]: :+1:

File: src/baldr/datetime.cc [hunk 2]
```diff
@@ -394,34 +393,118 @@ uint64_t seconds_since_epoch(const std::string& date_time,
...
+                     const uint64_t origin_seconds, const uint64_t dest_seconds,
+                     const boost::local_time::time_zone_ptr& origin_tz,
+                     const boost::local_time::time_zone_ptr& dest_tz,
+                     std::string& iso_origin, std::string& iso_dest) {
 
-  std::string iso_date_time;
-  if (!time_zone)
-    return iso_date_time;
+  iso_origin.empty();
+  iso_dest.empty();
```
Comment (dnesbitt61) [inline src/baldr/datetime.cc]: What do these do? Do you mean to clear the strings?

Comment (gknisely) [inline src/baldr/datetime.cc]: yes fixing

**Comment by gknisely** [comment]
@dnesbitt61 updated

**Comment by dnesbitt61** [comment]
:shipit:

**Pull request merged**